### PR TITLE
docs: Document local setup and debugging with VSCode

### PR DESCRIPTION
**Pull request by MonkeyDo**
### Problem
image

### Solution
Reworked the instructions, and published .vscode files 

### Areas of Impact
Just the markdown repo docs

### CODE REVIEW

**Comment by MonkeyDo** [comment]
@sbvkrishna and @panwarabhishek345 , considering you both have fresh eyes on this, and are both interested in the content, could you please have a look at this PR and tell me if anything needs to be changed improved?

**Comment by panwarabhishek345** [comment]
@MonkeyDo Will have a look at it soon.

File: NODEJS_SETUP.md [hunk 1]
```diff
@@ -21,93 +21,63 @@ This command will also compile the site LESS and JavaScript source files.
...
+services:
+  bookbrainz-site:
+  # 1. Change the command to run
+    command: npm run debug
+    volumes:
+      - "./config/config.json:/home/bookbrainz/bookbrainz-site/config/config.json:ro"
+  # 2. Mount the src directory
+      - "./src:/home/bookbrainz/bookbrainz-site/src"
+```
+**Note**: Using Webpack watch mode (`npm run debug`) results in more resource consumption (about ~1GB increased RAM usage) compared to running the [standard web server](#running-the-web-server).
```
Comment (sbvkrishna) [inline NODEJS_SETUP.md]: Just a small change
```suggestion
**Note**: Using Webpack watch mode (`npm run debug`) results in more resource consumption (about ~1GB increased RAM usage) compared to running the [standard web server](/README.md#running-the-web-server).
```

Comment (MonkeyDo) [reply NODEJS_SETUP.md]: Of course, thanks for catching that !

**Comment by sbvkrishna** [comment]
Everything looks great! 👌

**Pull request merged**